app/questionnaire/rules.py

```diff
@@ -231,20 +216,20 @@ def evaluate_when_rules(when_rules, schema, metadata, answer_store, routing_path
 
 
 def get_answer_store_value(answer_id, answer_store, schema, routing_path=None):
+    """ Return answer value assuming it is on the routing path.
+    If answer is not on the routing path, return None
+    If routing_path empty, return answer value
```

Comment:
Why do we still return if `routing_path=None`?

Reply:
I think so? Caught by the else I think?

Reply:
I don't think we ever want to return a value without checking if it's on the path. It could lead to bugs with returning or routing on values not on the path

Reply:
Ah, sorry, I misread. Yeah, makes sense.